### PR TITLE
Hook api.wmo into id.wmo

### LINKED ISSUE #51: Auto create a user when a valid token is used to create a project
As long as the API token is valid, we can create a new user in the db with it.

Follow up to #48

### PR DESCRIPTION
**Pull request by cadecairos**
fixes #51 

* [x] validate tokens with id.wmo
* [x] auto create user cols from api token data
* [x] tests

### CODE REVIEW

**Comment by cadecairos** [comment]
@campd whoops, accidentally assigned you haha. I got this :+1:

**Comment by cadecairos** [comment]
Just remembered that I need to modify the create user route to use the webmaker uid from the token, and not from a payload param.

**Comment by cadecairos** [comment]
@thisandagain @k88hudson this patch might break the API integration with webmaker-android

correct me if I'm wrong.

File: test/auth-config.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ var Lab = require('lab'),
   test = lab.test,
   expect = require('code').expect;
 
-var auth = require('../lib/auth-config');
+var auth = require('../lib/auth-config')(function() {});
```
Comment (thisandagain) [inline test/auth-config.js]: This is a little odd. Why are we passing an empty function here? It might be better to make the argument optional.

**Comment by thisandagain** [comment]
Looks great and tests well locally. One minor issue with the interface on the `auth-config` module but aside from that R+. :+1:

File: test/auth-config.js (hunk 1 above)
Comment (cadecairos) [inline test/auth-config.js]: This is just to ensure the hapi bearer token plugin doesn't throw. These tests only validate the auth config settings, and never calls the validateFunc

Comment (thisandagain) [inline test/auth-config.js]: Ok. Perhaps add a comment?

**Pull request merged**